api/clients/codecs/blob_test.go

```diff
@@ -0,0 +1,39 @@
+package codecs
+
+import (
+	"testing"
+
+	"github.com/Layr-Labs/eigenda/common/testutils/random"
+	"github.com/stretchr/testify/require"
+)
+
+// TestBlobConversion checks that internal blob conversion methods produce consistent results
+func TestBlobConversion(t *testing.T) {
+	testRandom := random.NewTestRandom(t)
+
+	iterations := 1000
+
+	for i := 0; i < iterations; i++ {
+		originalData := testRandom.Bytes(testRandom.Intn(1024) + 1)
+		testBlobConversionForForm(t, originalData, PolynomialFormEval)
+		testBlobConversionForForm(t, originalData, PolynomialFormCoeff)
+	}
+}
+
+func testBlobConversionForForm(t *testing.T, payloadBytes []byte, form PolynomialForm) {
+	payload := NewPayload(payloadBytes)
+
+	blob, err := payload.ToBlob(form)
```

Comment:
The natural interpretation for this reads wrong :(
It reads "payload is getting to blob in this form" instead of "payload, interpreted in this form, gets converted to a blob".
Maybe there's a different way to structure the method such that the interpretation makes more sense?

Reply:
Like perhaps defining on Blob a method `.fromPayloadInForm()`? Then we could use like
`new(Blob).fromPayloadInForm(payload, form)`. It's def more ugly but at least it reads more easily I feel?

Reply:
What do you think of this idea: when constructing a payload, declare in the constructor what form it is. The sequence would then read:

```
payload := NewPayload(payloadBytes, form)
blob, err := payload.ToBlob()
```

I personally *really* dislike like the pattern of creating a new empty object, just to call a method that creates _another_ instance of the object

Reply:
> just to call a method that creates another instance of the object
In this case you would call a method which changes the object in place though.

`payloadForm` name does remove the ambiguity though, thanks!